src/servers/src/http/table_result.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp::max;
+use std::fmt::Write;
+
+use axum::http::{header, HeaderValue};
+use axum::response::{IntoResponse, Response};
+use common_error::status_code::StatusCode;
+use common_query::Output;
+use itertools::Itertools;
+use mime_guess::mime;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::http::error_result::ErrorResponse;
+use crate::http::header::{GREPTIME_DB_HEADER_EXECUTION_TIME, GREPTIME_DB_HEADER_FORMAT};
+use crate::http::{handler, GreptimeQueryOutput, HttpResponse, ResponseFormat};
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct TableResponse {
+    output: Vec<GreptimeQueryOutput>,
+    execution_time_ms: u64,
+}
+
+impl TableResponse {
+    pub async fn from_output(outputs: Vec<crate::error::Result<Output>>) -> HttpResponse {
+        match handler::from_output(ResponseFormat::Csv, outputs).await {
+            Err(err) => HttpResponse::Error(err),
+            Ok((output, _)) => {
+                if output.len() > 1 {
+                    HttpResponse::Error(ErrorResponse::from_error_message(
+                        ResponseFormat::Table,
+                        StatusCode::InvalidArguments,
+                        "Multi-statements are not allowed".to_string(),
+                    ))
+                } else {
+                    HttpResponse::Table(TableResponse {
+                        output,
+                        execution_time_ms: 0,
+                    })
+                }
+            }
+        }
+    }
+
+    pub fn output(&self) -> &[GreptimeQueryOutput] {
+        &self.output
+    }
+
+    pub fn with_execution_time(mut self, execution_time: u64) -> Self {
+        self.execution_time_ms = execution_time;
+        self
+    }
+
+    pub fn execution_time_ms(&self) -> u64 {
+        self.execution_time_ms
+    }
+    fn into_payload(&mut self) -> String {
```

Comment:
prefer
```suggestion
    fn into_string(self) -> String {
```
Or 

```suggestion
    fn to_string(&self) -> String {
```

See also: https://pingcap.github.io/style-guide/rust/naming.html

BTW, The #3542 fixes a bug in the main branch. Rebase the main could reduce some strange fails.

Reply:
I tried, but it is better to impl Display trait. 
Because of [this](https://rust-lang.github.io/rust-clippy/master/index.html#/inherent_to_string)

Reply:
Maybe we can add `cargo clippy` command to pre-commit